pkg/provider/kubernetes/crd/kubernetes_http.go

```diff
@@ -332,12 +340,12 @@ func (c configBuilder) loadServers(fallbackNamespace string, svc v1alpha1.LoadBa
 		}
 
 		if port == 0 {
-			return nil, fmt.Errorf("cannot define a port for %s/%s", namespace, sanitizedName)
+			return nil, false, fmt.Errorf("cannot define a port for %s/%s", namespace, sanitizedName)
 		}
 
 		protocol, err := parseServiceProtocol(svc.Scheme, svcPort.Name, svcPort.Port)
```

Comment:
I think, in the first place, this should not have been inside this for loop at all. Not your fault of course. Should we move it outside of the for loop? Maybe you also want to update your PR to the upstream.